src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -899,14 +688,14 @@ private void ValidateTransition(BatchWriterState newState)
 
                     break;
                 case BatchWriterState.BatchStarted:
-                    if (newState != BatchWriterState.ChangeSetStarted && newState != BatchWriterState.OperationCreated && newState != BatchWriterState.BatchCompleted)
+                    if (newState != BatchWriterState.ChangesetStarted && newState != BatchWriterState.OperationCreated && newState != BatchWriterState.BatchCompleted)
                     {
                         throw new ODataException(Strings.ODataBatchWriter_InvalidTransitionFromBatchStarted);
                     }
 
                     break;
-                case BatchWriterState.ChangeSetStarted:
-                    if (newState != BatchWriterState.OperationCreated && newState != BatchWriterState.ChangeSetCompleted)
+                case BatchWriterState.ChangesetStarted:
+                    if (newState != BatchWriterState.OperationCreated && newState != BatchWriterState.ChangesetCompleted)
                     {
                         throw new ODataException(Strings.ODataBatchWriter_InvalidTransitionFromChangeSetStarted);

```

Comment:
It looks like we decided to change the letter casing for "Change**S**etStarted" to "Change**s**etStarted". We should probably do a search across the project for other variables that have "ChangeSet" and update them for consistency.

ODataBatchWriter_InvalidTransitionFromChange**S**etStarted